pkg/upstream/manager.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upstream
+
+import (
+	"context"
+	"sync"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/pkg/config"
+	cerror "github.com/pingcap/tiflow/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// DefaultClusterID is a pseudo cluster id for now. It will be removed in the future.
+const DefaultClusterID uint64 = 0
+
+// UpManager is a global variable.
+var UpManager *Manager
```

Comment:
- Making it a global variable might cause difficult when we go serverless with Dataflow Engine.
- If we do choose to use a global variable, please make sure that (1) It is clear where, when and how it is initialized, and (2) It is exposed through a function `GlobalUpstreamManager() *Manager`, and the actual global variable is made private to prevent accidental modification of the pointer.